src/sqlfluff/dialects/dialect_mysql.py

```diff
@@ -90,3 +97,45 @@ class CreateTableStatementSegment(
             ),
         ],
     )
+
+mysql_dialect.add(
+    DoubleForwardSlashSegment=StringParser(
+        "//", SymbolSegment, name="doubleforwardslash", type="statement_terminator"
+    ),
+    DoubleDollarSignSegment=StringParser(
+        "$$", SymbolSegment, name="doubledollarsign", type="statement_terminator"
+    ),
+)
+
+mysql_dialect.replace(
+    DelimiterSegment=OneOf(Ref("SemicolonSegment"), Ref("TildeSegment")),
+    TildeSegment=StringParser(
+        "~", SymbolSegment, name="tilde", type="statement_terminator"
+    ), 
+)
+
+@mysql_dialect.segment()
+class DelimiterGrammar(BaseSegment):
+    match_grammar = StartsWith("DELIMITER")
+    parse_grammar = Sequence(
+        "DELIMITER",
+        OneOf(Ref("DelimiterSegment")),
+        optional=True,
```

Comment:
Is it necessary to specify `optional=True` here?

Reply:
I did that because delimiter isn't always specified.  If you don't specify the `DELIMTER` keyword then it just assumes `;`